### DISCUSSION TITLE
How to get repo contents with latetst commit for every file?

### DISCUSSION THREAD

**Discussion by RealPeha** [octokit/octokit.js#2021] [Q&A] [answered]
Hello everyone!

Please, tell me how do i get all this data for the repository?

![image](https://user-images.githubusercontent.com/26817340/108889405-0417ee00-7615-11eb-8aeb-dd100c6b9551.png)

I know about `octokit.repos.getContent` but this endpoint returns me only files and dirs, what about commits?

**Comment by gr2m** [comment, marked as answer]
There might be a way to do this using GraphQL: https://docs.github.com/en/graphql. A good place to ask such a question is https://github.community/c/github-ecosystem/37

**Reply by RealPeha** [reply]
Cool, I'll try it through GraphQL

Thanks for the links!
